File: Unidad 2 Numeros Pseudoaleatorios/NumerosPseudoaleatorios2.py
import random

def generar_numeros_aleatorios(cantidad, semilla=123):
    random.seed(semilla)
    lista_numeros = []

    for _ in range(cantidad):
        numero = random.random() * (20 - 5) + 5  # Escala a [5, 20]
        lista_numeros.append(numero)

    return lista_numeros

# Ejecución
total = 10
numeros = generar_numeros_aleatorios(total)
print(f"Lista de {total} números pseudoaleatorios entre 5 y 20:")
for i, num in enumerate(numeros, start=1):
    print(f"{i:02d}: {num:.5f}")

File: Unidad 2 Numeros Pseudoaleatorios/NumerosPseudoAleatorios2.java
import java.util.Random;

public class NumerosPseudoAleatorios2 {
    public static void main(String[] args) {
        Random generador = new Random(456); // Semilla fija para reproducibilidad
        int cantidad = 20;

        System.out.println("Generando " + cantidad + " números pseudoaleatorios entre 0 y 1:");

        for (int i = 0; i < cantidad; i++) {
            double numero = generador.nextDouble();
            System.out.printf("Número %2d: %.6f%n", i + 1, numero);
        }
    }
}
